code/modules/heavy_vehicle/mech_interaction.dm

```diff
@@ -565,3 +541,163 @@
 		src.visible_message("<span class='warning'>\The [src] beeps loudly as its servos sieze up, and it enters lockdown mode!</span>")
 	else
 		src.visible_message("<span class='warning'>\The [src] hums with life as it is released from its lockdown mode!</span>")
+
+/mob/living/heavy_vehicle/get_floating_chat_x_offset()
+	return 8
+
+/mob/living/heavy_vehicle/hear_say(var/message, var/verb = "says", var/datum/language/language = null, var/alt_name = "", var/italics = 0, var/mob/speaker = null, var/sound/speech_sound, var/sound_vol)
+	if(can_listen())
+		handle_hear_say(speaker, message)
+	return ..()
+
+// heavily commented so it doesn't look like one fat chunk of code, which it still does - Geeves
+/mob/living/heavy_vehicle/proc/handle_hear_say(var/mob/speaker, var/text)
+	var/found_text = findtext(text, name)
+	if(!found_text)
+		found_text = findtext(text, nickname)
+	if(found_text)
+		text = copytext(text, found_text) // I'm trimming the text each time so only information stated after eachother is valid
+		sleep(5)
```

Comment:
Ewwww. Maybe make the post-sleep part in `/proc/respond_to_command()` or something, and use a timer?

Reply:
made it work on a timer